nifi-nar-bundles/nifi-smb-bundle/nifi-smb-processors/src/main/java/org/apache/nifi/processors/smb/PutSmbFile.java

```diff
@@ -0,0 +1,334 @@
+        final String username = context.getProperty(USERNAME).getValue();
+        String password = context.getProperty(PASSWORD).getValue();
+
+        AuthenticationContext ac = null;
+        if (username != null && password != null) {
+            ac = new AuthenticationContext(
+                username,
+                password.toCharArray(),
+                domain);
+        } else {
+            ac = AuthenticationContext.anonymous();
+        }
+
+        try (Connection connection = smbClient.connect(hostname);
+            Session smbSession = connection.authenticate(ac);
+            DiskShare share = (DiskShare) smbSession.connectShare(shareName)) {
+
+            for (FlowFile flowFile : flowFiles) {
+                String directory = context.getProperty(DIRECTORY).evaluateAttributeExpressions(flowFile).getValue();
+                final String filename = flowFile.getAttribute(CoreAttributes.FILENAME.key());
+                final long sendStart = System.nanoTime();
+                String fullPath;
+
+                if (directory == null) {
+                    directory = "";
+                    fullPath = filename;
+                } else {
+                    fullPath = directory + "\\" + filename;
+
+                    // missing directory handling
+                    if (context.getProperty(CREATE_DIRS).asBoolean() && !share.folderExists(directory)) {
+                        logger.debug("Creating folder {}", new Object[]{directory});
+                        share.mkdir(directory);
+                    }
+                }
+
+                final URI uri = new URI("smb", hostname, "/" + fullPath.replace('\\', '/'), null);
+
+                // replace strategy handling
+                SMB2CreateDisposition createDisposition = SMB2CreateDisposition.FILE_OVERWRITE_IF;
+                final String conflictResolution = context.getProperty(CONFLICT_RESOLUTION).getValue();
+                if (!conflictResolution.equals(REPLACE_RESOLUTION) && share.fileExists(fullPath)) {
+                    if (conflictResolution.equals(IGNORE_RESOLUTION)) {
+                        session.transfer(flowFile, REL_SUCCESS);
+                        logger.info("Transferring {} to success because file with same name already exists", new Object[]{flowFile});
+                        continue;
+                    } else if (conflictResolution.equals(FAIL_RESOLUTION)) {
+                        flowFile = session.penalize(flowFile);
+                        logger.warn("Penalizing {} and routing to failure as configured because file with the same name already exists", new Object[]{flowFile});
+                        session.transfer(flowFile, REL_FAILURE);
+                        continue;
+                    }
+                }
+
+
+                try (File f = share.openFile(
+                        fullPath,
+                        EnumSet.of(AccessMask.GENERIC_ALL),
```

Comment:
It is better to require only writing rights using GENERIC_WRITE rather than GENERIC_ALL.
The mask GENERIC_ALL implies execution rights which are not mandatory to put files. 
This can lead to denied connection if the samba share does not allow the user to execute.

Thank you for your work on this, it is really usefull.

Reply:
ok, update to GENERIC_WRITE